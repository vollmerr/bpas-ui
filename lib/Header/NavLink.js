var _templateObject = _taggedTemplateLiteral(['\n    text-align: ', ';\n    flex: 1;\n    list-style-type: none;\n\n    &:not(:last-child) {\n      border-bottom: ', ';\n    }\n'], ['\n    text-align: ', ';\n    flex: 1;\n    list-style-type: none;\n\n    &:not(:last-child) {\n      border-bottom: ', ';\n    }\n']),
    _templateObject2 = _taggedTemplateLiteral(['\n    display: ', ';\n    vertical-align: ', ';\n    color: ', ';\n    padding-right: ', ';\n'], ['\n    display: ', ';\n    vertical-align: ', ';\n    color: ', ';\n    padding-right: ', ';\n']),
    _templateObject3 = _taggedTemplateLiteral(['\n    display: block;\n    color: inherit;\n    padding: ', 'px ', 'px;\n\n    &:hover,\n    &:focus,\n    &.active,\n    &:hover > ', ',\n    &:focus > ', ',\n    &.active > ', ' {\n      text-decoration: none;\n      color: ', ';\n    }\n'], ['\n    display: block;\n    color: inherit;\n    padding: ', 'px ', 'px;\n\n    &:hover,\n    &:focus,\n    &.active,\n    &:hover > ', ',\n    &:focus > ', ',\n    &.active > ', ' {\n      text-decoration: none;\n      color: ', ';\n    }\n']);

function _taggedTemplateLiteral(strings, raw) { return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); }

import React from 'react';
import PropTypes from 'prop-types';
import { NavLink as Link } from 'react-router-dom';
import styled from 'styled-components';
import Icon from '../Icon';
import theme from '../util/theme';

var Wrapper = styled.li(_templateObject, function (props) {
  return props.isMobile ? 'left' : 'center';
}, function (props) {
  return props.isMobile ? '1px solid ' + theme.color.theme.tertiary : 'none';
});

var StyledIcon = styled(Icon)(_templateObject2, function (props) {
  return props.isMobile ? 'inline-block' : 'block';
}, function (props) {
  return props.isMobile ? 'middle' : 'baseline';
}, theme.color.theme.tertiary, function (props) {
  return props.isMobile ? '10px' : '0';
});

var StyledLink = styled(Link)(_templateObject3, theme.size.sm, theme.size.xl, StyledIcon, StyledIcon, StyledIcon, theme.color.theme.primary);

/** Renders a single navbar link */
function NavLink(_ref) {
  var link = _ref.link,
      isMobile = _ref.isMobile,
      onClick = _ref.onClick;

  return React.createElement(
    Wrapper,
    { isMobile: isMobile },
    React.createElement(
      StyledLink,
      {
        exact: link.href === '/',
        to: link.href,
        onClick: onClick
      },
      React.createElement(StyledIcon, {
        icon: link.icon,
        size: 32,
        isMobile: isMobile
      }),
      link.text
    )
  );
}

NavLink.PropTypes = {
  link: PropTypes.shape({
    text: PropTypes.string.isRequired,
    href: PropTypes.string.isRequired,
    icon: PropTypes.string.isRequired
  }).isRequired,
  isMobile: PropTypes.bool,
  onClick: PropTypes.func
};

export default NavLink;