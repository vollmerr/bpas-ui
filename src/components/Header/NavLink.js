import React from 'react';
import PropTypes from 'prop-types';
import { NavLink as Link } from 'react-router-dom';
import styled from 'styled-components';
import Icon from '../Icon';
import theme from '../../util/theme';

const Wrapper = styled.li`
    text-align: ${props => props.isMobile ? 'left' : 'center'};
    flex: 1;

    &:not(:last-child) {
      border-bottom: ${props => props.isMobile ? `1px solid ${theme.color.primaryFaded}` : 'none'};
    }
`;

const StyledIcon = styled(Icon)`
    display: ${props => props.isMobile ? 'inline-block' : 'block'};
    vertical-align: ${props => props.isMobile ? 'middle' : 'baseline'};
    color: ${theme.color.primaryFaded};
    padding-right: ${props => props.isMobile ? '10px' : '0'};
`;

const StyledLink = styled(Link)`
    display: block;
    color: inherit;
    padding: ${theme.size.sm}px ${theme.size.xl}px;

    &:hover,
    &:focus,
    &.active,
    &:hover > ${StyledIcon},
    &:focus > ${StyledIcon},
    &.active > ${StyledIcon} {
      text-decoration: none;
      color: ${theme.color.primary};
    }
`;

/** Renders a single navbar link */
function NavLink({
  link,
  isMobile,
  onClick,
}) {
  return (
    <Wrapper isMobile={isMobile}>
      <StyledLink
        exact={link.href === '/'}
        to={link.href}
        onClick={onClick}
      >
        <StyledIcon
          icon={link.icon}
          size={32}
          isMobile={isMobile}
        />
        {link.text}
      </StyledLink>
    </Wrapper>
  );
}

NavLink.PropTypes = {
  link: PropTypes.shape({
    text: PropTypes.string.isRequired,
    href: PropTypes.string.isRequired,
    icon: PropTypes.string.isRequired,
  }).isRequired,
  isMobile: PropTypes.bool,
  onClick: PropTypes.func,
};

export default NavLink;
