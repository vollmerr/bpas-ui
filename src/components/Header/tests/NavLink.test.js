import React from 'react';
import { shallow } from 'enzyme';
import NavLink from '../NavLink';

const testLink = {
  text: 'test text',
  href: '/test-href',
  icon: 'home',
};

describe('<NavLink />', () => {
  it('renders correctly', () => {
    const tree = shallow(
      <NavLink link={testLink} />,
      { context: {history: {}} }
    );
    expect(tree).toMatchSnapshot();
  });

  xit('adds the active class when active', () => {});

  xit('passes the correct props down', () => {});
});
