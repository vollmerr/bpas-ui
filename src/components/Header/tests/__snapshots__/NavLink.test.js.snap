// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavLink /> renders correctly 1`] = `
<styled.li>
  <Styled(NavLink)
    exact={false}
    to="/test-href"
  >
    <Styled(Icon)
      icon="home"
      size={32}
    />
    test text
  </Styled(NavLink)>
</styled.li>
`;
